common/models/model-method.js

```diff
@@ -10,4 +11,38 @@ module.exports = function(ModelMethod) {
    * @class ModelMethod
    * @inherits WorkspaceEntity
    */
+
+  ModelMethod._shouldEncodeStaticFlagInName = function() {
+    var version = ModelMethod.app.models.Workspace.loopBackVersion;
+    return encodeStaticFlagInName(version);
+  };
+
+  ModelMethod.getJsonKey = function(data) {
+    if (this._shouldEncodeStaticFlagInName()) {
+      var isStatic = data.isStatic;
```

Comment:
In order to reduce nesting/indentation level, it's better to use reverse logic + early return. It also makes the code easier to read, as one can quickly determine what the other branch does and forget about it. In your version, one has to keep in mind that they need to check what happens when static flag is not encoded after they finish examination of the branch when the flag is encoded.

```js
{
  if (!this._shouldEncodeStaticFlagInName())
    return data.name;

  var isStatic = data.isStatic;
  // ...
}
```